Fix #20601 - Fixes translation suggestion handling by emitting from the modal and moving logic to the CARS component.

Ash-2k3 opened the pull request:
## Overview

1. This PR fixes or fixes part ofhttps://github.com/oppia/oppia/issues/20601
2. The original bug occurred because we were maintaining the queuedSuggestion in the translation modal. But once the modal gets closed, it looses the hold of the queuedSuggestion. Also there were some race conditions because of which network calls to accpet/reject suggestions were being made more than once for same queuedSuggestion. This has been taken care of by adding required conditional statements.

To resolve the issue, the translation review modal will now emit the queued suggestion instead of holding it. Rest of the processing will be taken care by cars component.

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [x] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [x] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [x] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).

## Proof that changes are correct

### Before Changes

![image](https://github.com/user-attachments/assets/f5f048d5-02a2-4856-a4cf-3cf8b9ce7586)

### After Changes

https://github.com/user-attachments/assets/e0c3e168-b6bf-4f31-9465-7e3b61f9989c



## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

Ash-2k3 committed:
Initial commit for undo feat bug fix

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Ash-2k3 committed:
Testing unit tests

Ash-2k3 committed:
Fix unit tests

Ash-2k3 committed:
Fix unit tests for cars components

Ash-2k3 committed:
Fix unit tests for cars components

Ash-2k3 committed:
Fix unit tests for cars components

Ash-2k3 committed:
Fix undo feature

Ash-2k3 committed:
Fix unit tests

Ash-2k3 committed:
Run prettier

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Ash-2k3 committed:
Fix linter checks

Ash-2k3 committed:
Merge branch 'bug-pinunpin' of https://github.com/Ash-2k3/oppia into bug-pinunpin

Ash-2k3 committed:
Fix unit tests

Ash-2k3 committed:
Fix unit tests

Ash-2k3 committed:
Fix failing tests

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Ash-2k3 committed:
Fix failing tests

Ash-2k3 committed:
Merge branch 'bug-pinunpin' of https://github.com/Ash-2k3/oppia into bug-pinunpin

Ash-2k3 committed:
Fix failing tests

Ash-2k3 committed:
Fix failing tests

Ash-2k3 committed:
Fix failing tests

Ash-2k3 committed:
Fix failing tests

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Ash-2k3 commented:
TODO - I have to attach the proof of working. I will record the screen and post it tomorrow.

masterboy376 commented on core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.spec.ts:
```diff
@@ -2589,4 +2543,245 @@ describe('Contributions and review component', () => {
+    });
+
+    it(
+      'should commit the queued suggestion when' + ' the snackbar is dismissed',
```
Why concatenating string here?

Ash-2k3 replied:
Fixed this, thanks for catching it.

masterboy376 reviewed (changes requested):
Thanks @Ash-2k3! 
Looks fine, just a single comment though. PTAL

masterboy376 commented:
@Nik-09 PTAL!

Nik-09 reviewed (changes requested):
Please do the following and reassign me:

1. Update the title of the PR
2. Tick on the essential checklist of the PR description
3. Add video proof of "before" and "after" changes.

Thanks!

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

masterboy376 commented:
@Ash-2k3 PTAL

Ash-2k3 committed:
Update contributions-and-review.component.spec.ts

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Ash-2k3 commented:
@masterboy376 @Nik-09 PTAL, I have addressed your comments. @Nik-09 I am not able to launch my dev server because of system resources issue. I have attached the screenshot from the issue itself for now.

masterboy376 reviewed (approved):
Thanks @Ash-2k3, LGTM! Just a note, could you check the CI checks.

Ash-2k3 committed:
Prettier changes

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

Nik-09 commented:
Hi @Ash-2k3, sorry for the delay, I will review this in 1-2 hours. Thank you!

Nik-09 reviewed (changes requested):
One question in the video, after accepting the suggestion (0:18 sec) the "suggestion in queue" section is shown in the bottom right. At 0:29 sec the suggestion is accepted. But in the bottom right, "suggestion in queue" section is still showing. What happens if I click "undo" at that time?

Ash-2k3 commented:
@Nik-09, If you notice, at 0:17, I accepted a suggestion with the translated content "3", and the next second I accepted another suggestion with the translated content "6". Since we accepted two suggestions in a row, "3" would have been committed to the backend immediately, and the queued one would be "6".

Now, due to some network delays, the confirmation snackbar for accepting "3" appeared late (around 0:28). At this point, the snackbar is holding the suggestion "6", so if you click undo, it will undo the suggestion "6".

Ash-2k3 committed:
Merge branch 'develop' into bug-pinunpin

masterboy376 reviewed (approved):
LGTM
